mlflow/langchain/__init__.py

```diff
@@ -863,16 +856,21 @@ def _load_pyfunc(path: str, model_config: Optional[Dict[str, Any]] = None):
 
 def _load_model_from_local_fs(local_model_path):
     flavor_conf = _get_flavor_configuration(model_path=local_model_path, flavor_name=FLAVOR_NAME)
-    if MODEL_CODE_PATH in flavor_conf:
-        model_config = flavor_conf.get(MODEL_CONFIG, None)
+    pyfunc_flavor_conf = _get_flavor_configuration(
+        model_path=local_model_path, flavor_name=PYFUNC_FLAVOR_NAME
+    )
+    if MODEL_CODE_PATH in pyfunc_flavor_conf or MODEL_CODE_PATH in flavor_conf:
+        model_config = flavor_conf.get(MODEL_CONFIG, pyfunc_flavor_conf.get(MODEL_CONFIG, None))
         if isinstance(model_config, str):
             config_path = os.path.join(
                 local_model_path,
                 os.path.basename(model_config),
             )
             model_config = _load_from_yaml(config_path)
 
-        flavor_code_path = flavor_conf.get(MODEL_CODE_PATH)
+        flavor_code_path = pyfunc_flavor_conf.get(MODEL_CODE_PATH) or flavor_conf.get(
+            MODEL_CODE_PATH
+        )
```

Comment:
On line 863, we load the flavor conf preferentially. Here, we load the pyfunc conf preferentially. Should we be consistent: always load flavor conf if present, else load pyfunc conf?